group-coordinator/src/test/java/org/apache/kafka/coordinator/group/runtime/MultiThreadedEventProcessorTest.java

```diff
@@ -53,53 +50,19 @@
 
 @Timeout(value = 60)
 public class MultiThreadedEventProcessorTest {
-    private static class MockEventAccumulator<T> extends EventAccumulator<TopicPartition, CoordinatorEvent> {
+    private static class DelayEventAccumulator extends EventAccumulator<TopicPartition, CoordinatorEvent> {
         private final Time time;
-        private final Queue<CoordinatorEvent> events;
-        private final long timeToPollMs;
-        private final AtomicBoolean isClosed;
+        private final long takeDelayMs;
 
-        public MockEventAccumulator(Time time, long timeToPollMs) {
+        public DelayEventAccumulator(Time time, long takeDelayMs) {
             this.time = time;
-            this.events = new LinkedList<>();
-            this.timeToPollMs = timeToPollMs;
-            this.isClosed = new AtomicBoolean(false);
+            this.takeDelayMs = takeDelayMs;
         }
 
         @Override
-        public CoordinatorEvent poll() {
-            synchronized (events) {
```

Comment:
It's interesting that we basically made the real accumulator do what this mock was doing 😅 
But it also makes sense 👍

Reply:
I think that it was already doing it. I am not sure why we implemented it this way.